app/jinja_filters.py

```diff
@@ -35,7 +35,7 @@ def format_number(value: Union[int, Decimal, float]) -> str:
     formatted_number: str
     if value or value == 0:
         formatted_number = numbers.format_decimal(
-            value, locale=flask_babel.get_locale()
+            value, locale=flask_babel.get_locale(), decimal_quantization=False
```

Comment:
`decimal_quantization` doesn't fix it for leading zeros. If a user enters 1.15000, it will render 1.15 even though we store 1.15000 and send downstream 1.15000.